packages/cli/lib/services/compile.service.ts

```diff
@@ -206,16 +231,23 @@ async function compile({
     const fileNameWithExtension = `${basename}-${providerConfiguration.providerConfigKey}${extname}`;
     const outputPath = path.join(dirname, fileNameWithExtension);
 
+    if (debug) {
+        printDebug(`Compiling ${file.inputPath} -> ${outputPath}`);
+    }
+
     await build({
         entryPoints: [file.inputPath],
-        tsconfig: getNangoRootPath() + '/tsconfig.dev.json',
+        tsconfig: path.join(getNangoRootPath()!, 'tsconfig.dev.json'),
         skipNodeModulesBundle: true,
         silent: !debug,
-        outDir: dirname,
+        outDir: path.join(fullPath, 'dist'),
+        outExtension: () => ({ js: '.js' }),
```

Comment:
I realized when testing more deeply that if you had a package.json with type:module it would compile to cjs instead so I'm forcing the extension